### ISSUE TITLE
Mistakenly including cached_property as a field

### ISSUE THREAD

**Issue by denis-manakov-cndt** [pydantic/pydantic#4609]
### Initial Checks

- [X] I have searched GitHub for a duplicate issue and I'm sure this is something new
- [X] I have searched Google & StackOverflow for a solution and couldn't find anything
- [X] I have read and followed [the docs](https://pydantic-docs.helpmanual.io) and still think this is a bug
- [X] I am confident that the issue is with pydantic (not my code, or another library in the ecosystem like [FastAPI](https://fastapi.tiangolo.com) or [mypy](https://mypy.readthedocs.io/en/stable))

### Description

Python 3.9.13
pydantic  1.9.1 and 1.10.2 (both version affected)

after first call cached_property creates field in model __dict__, then pydantic starts including this field in .dict() serialization. Insertion always happens, regardless on Config: extra = Extra.forbid/ignore and regardless on property name - property with name __foo still affected.

### Example Code

```Python
from functools import cached_property

from pydantic import __version__, BaseModel

print(__version__)


class A(BaseModel):
    a: int

    class Config:
        keep_untouched = (cached_property,)

    @cached_property
    def foo(self):
        print('calculated')
        return lambda x: self.a # Some unseriliazable result to trigger exception


# Ok case
obj = A(a=1)
print(obj.dict())
# {'a': 1}
print(obj.json())
# "{"a": 1}"

# Exception after obj.foo call
obj = A(a=1)
print(obj.foo) # Internally modifies obj.__dict__
print(obj.dict())
# Expected: {'a': 1}
# Actual: {'a': 1, 'foo': <function A.foo.<locals>.<lambda> at 0x7fa648032160>} # Whoops, foo in dict 🙄🙄🙄
print(obj.json())
# Expected: "{"a": 1}"
# Actual: Exception
```

### Python, Pydantic & OS Version

```Text
pydantic version: 1.9.1 
            pydantic compiled: True
                 install path: /home/manakov_dev/.cache/pypoetry/virtualenvs/etl-u0m7Z2iP-py3
.9/lib/python3.9/site-packages/pydantic
               python version: 3.9.13 (main, Aug  8 2022, 20:56:30)  [GCC 9.4.0]
                     platform: Linux-4.4.0-19041-Microsoft-x86_64-with-glibc2.31
     optional deps. installed: ['typing-extensions']
```

### Affected Components

- [ ] [Compatibility between releases](https://pydantic-docs.helpmanual.io/changelog/)
- [ ] [Data validation/parsing](https://pydantic-docs.helpmanual.io/usage/models/#basic-model-usage)
- [X] [Data serialization](https://pydantic-docs.helpmanual.io/usage/exporting_models/) - `.dict()` and `.json()`
- [ ] [JSON Schema](https://pydantic-docs.helpmanual.io/usage/schema/)
- [ ] [Dataclasses](https://pydantic-docs.helpmanual.io/usage/dataclasses/)
- [ ] [Model Config](https://pydantic-docs.helpmanual.io/usage/model_config/)
- [ ] [Field Types](https://pydantic-docs.helpmanual.io/usage/types/) - adding or changing a particular data type
- [ ] [Function validation decorator](https://pydantic-docs.helpmanual.io/usage/validation_decorator/)
- [ ] [Generic Models](https://pydantic-docs.helpmanual.io/usage/models/#generic-models)
- [ ] [Other Model behaviour](https://pydantic-docs.helpmanual.io/usage/models/) - `construct()`, pickling, private attributes, ORM mode
- [ ] [Plugins](https://pydantic-docs.helpmanual.io/) and integration with other tools - mypy, FastAPI, python-devtools, Hypothesis, VS Code, PyCharm, etc.

**Comment by denis-manakov-cndt** [comment]
Same with underscored property name:

```
class A(BaseModel):
    a: int

    class Config:
        keep_untouched = (cached_property,)

    @cached_property
    def __foo(self):
        print('calculated')
        return lambda x: self.a

    @property
    def foo(self):
        return self.__foo
```

**Comment by samuelcolvin** [comment]
We don't suggest anywhere that we support `cached_property`, this is unfortunate but an expected side effect of using `cached_property`.

I hope we can implemented computed fields #2625 in V2, which is partially related.

I really don't know what to do about this, we definitely don't want to filter `__dict__` on serialisation for performance reasons.

**Comment by skewty** [comment]
Similar issue with `functools.lru_cache`.
